tests/stamp_test.py

```diff
@@ -61,9 +61,12 @@ def check_mtime(self, mtime, file_path, file_name):
     if ((mtime < self.target_mtime - StampTest.ALLOWED_DELTA_FROM_NOW)
         or (mtime > self.target_mtime)):
        self.fail(
-           'Archive %s contains file %s with mtime:%d, expected:%d +/- %d' % (
+           'Archive %s contains file %s with mtime:%d, expected:%d +/- %d.' % (
                file_path, file_name, mtime, self.target_mtime,
-               StampTest.ALLOWED_DELTA_FROM_NOW))
+               StampTest.ALLOWED_DELTA_FROM_NOW) +
+           '  This may be a false positive if your build cache is more than' +
+           ' %s seconds old.  If so, try bazel clean and rerun the test.' %
```

Comment:
Could perhaps phrase this in terms of minutes or hours rather than seconds -- would require less thinking by those unfamiliar.

Not a critical problem, though.

Reply:
nano seconds!!!